test/core/security/local_security_connector_test.cc

```diff
@@ -0,0 +1,165 @@
+#include "test/core/test_util/tls_utils.h"
+#include "googletest/include/gtest/gtest.h"
+
+namespace grpc_core {
+namespace testing {
+
+class LocalSecurityConnectorTest : public ::testing::Test {
+ protected:
+  LocalSecurityConnectorTest() {}
+};
+
+static void me_read(grpc_endpoint* ep, grpc_slice_buffer* slices,
+                    grpc_closure* cb, bool urgent, int min_progress_size) {}
+
+static void me_write(grpc_endpoint* ep, grpc_slice_buffer* slices,
+                     grpc_closure* cb, void* arg, int max_frame_size) {}
+
+static void me_add_to_pollset(grpc_endpoint* /*ep*/,
+                              grpc_pollset* /*pollset*/) {}
+
+static void me_add_to_pollset_set(grpc_endpoint* /*ep*/,
+                                  grpc_pollset_set* /*pollset*/) {}
+
+static void me_delete_from_pollset_set(grpc_endpoint* /*ep*/,
+                                       grpc_pollset_set* /*pollset*/) {}
+
+static void me_destroy(grpc_endpoint* ep) {}
+
+static absl::string_view me_get_peer(grpc_endpoint* /*ep*/) {
+  return "";
+}
+
+static absl::string_view me_get_local_address_unix(grpc_endpoint* /*ep*/) {
+  return "unix:";
+}
+
+static int me_get_fd(grpc_endpoint* /*ep*/) { return -1; }
+
+static bool me_can_track_err(grpc_endpoint* /*ep*/) { return false; }
+
+static const grpc_endpoint_vtable vtable_unix = {me_read,
+                                            me_write,
+                                            me_add_to_pollset,
+                                            me_add_to_pollset_set,
+                                            me_delete_from_pollset_set,
+                                            me_destroy,
+                                            me_get_peer,
+                                            me_get_local_address_unix,
+                                            me_get_fd,
+                                            me_can_track_err};
+
+static absl::string_view me_get_local_address_local(grpc_endpoint* /*ep*/) {
+  return "ipv4:127.0.0.1:12667";
+}
+
+static const grpc_endpoint_vtable vtable_local = {me_read,
+                                            me_write,
+                                            me_add_to_pollset,
+                                            me_add_to_pollset_set,
+                                            me_delete_from_pollset_set,
+                                            me_destroy,
+                                            me_get_peer,
+                                            me_get_local_address_local,
+                                            me_get_fd,
+                                            me_can_track_err};
+
+static void check_tsi_security_level(grpc_local_connect_type connect_type,
+                                 tsi_security_level level, grpc_endpoint ep) {
+  grpc_server_credentials* server_creds = grpc_local_server_credentials_create(connect_type);
+  ChannelArgs args;
+  EXPECT_NE(server_creds, nullptr);
```

Comment:
Please either remove `EXPECT_NE` (because `grpc_local_server_credentials_create` never returns null) or change it to `ASSERT_NE`.

If we decide to leave it, please move it to the next line immediately after `grpc_local_server_credentials_create` is called.

Similarly for `EXPECT_NE(connector, nullptr);`

Reply:
Agreed for removing `EXPECT_NE(server_creds)`
For the connector one - I think technically you're right (server_creds is not null for this test) -
https://github.com/grpc/grpc/blob/b1896a2136e9f60e026daeaaf393cb8cb143be72/src/core/lib/security/security_connector/local/local_security_connector.cc#L298
but in such cases is it better to drop or to use `ASSERT_NE` ?

Reply:
Took a look at the code and it seems that `connector` could be nullptr, so let's keep the check and use `ASSERT_NE`.